laufband/graphband.py

```
@@ -26,77 +68,74 @@ class GraphTraversalProtocol(t.Protocol):
         *,
-        hash_fn: Callable[[_T], str] | None = None,
-        lock: Lock | None = None,
-        lock_path: Path | str | None = None,
-        com: Path | str | None = None,
-        identifier: str | t.Callable | None = None,
-        cleanup: bool = False,
-        failure_policy: t.Literal["continue", "stop"] = "continue",
-        heartbeat_timeout: int | None = None,
-        max_died_retries: int | None = None,
-        disable: bool | None = None,
+        lock: Lock = Lock("graphband.lock"),
```

Ruff (0.12.2)

Do not perform function call `Lock` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)